_tools/packages.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+import { walk } from "../fs/walk.ts";
+import { relative } from "../path/mod.ts";
+
+export async function discoverPackages() {
+  const packages = [];
+  for await (const entry of Deno.readDir(".")) {
+    if (
+      entry.isDirectory && !entry.name.startsWith(".") &&
+      !entry.name.startsWith("_") && entry.name !== "coverage"
+    ) {
+      packages.push(entry.name);
+    }
+  }
+  packages.sort();
+
+  console.log("Discovered", packages.length, "packages.");
+  return packages;
+}
+
+export async function discoverExportsByPackage(packages: string[]) {
+  // Collect all of the exports for each package.
+  const exportsByPackage = new Map<string, [string, string][]>();
+  for (const pkg of packages) {
+    const exports = await discoverExports(pkg);
+    exportsByPackage.set(pkg, exports);
+  }
+  return exportsByPackage;
+}
+
+async function discoverExports(pkg: string) {
+  const exports: [string, string][] = [];
+  const base = await Deno.realPath(pkg);
+  const files = walk(base);
```

Comment:
Nit: instead of filtering during loop iteration, we could provide `walk()` options that do that for us.

Reply:
Updated to filter like this:

```
    includeFiles: true,
    includeDirs: false,
    includeSymlinks: false,
```